cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go

```diff
@@ -12,3 +12,292 @@ limitations under the License.
+		t.Run(tc.name, func(t *testing.T) {
+			s := Server{clientGetter: tc.clientGetter, manager: tc.manager}
+
+			client, errClient := s.GetClient(context.Background())
+
+			if got, want := status.Code(errClient), tc.statusCodeClient; got != want {
+				t.Errorf("got: %+v, want: %+v", got, want)
+			}
+
+			_, errManager := s.GetManager()
+
+			if got, want := status.Code(errManager), tc.statusCodeManager; got != want {
+				t.Errorf("got: %+v, want: %+v", got, want)
+			}
+
+			// If there is no error, the client should be a dynamic.Interface implementation.
+			if tc.statusCodeClient == codes.OK {
+				if _, ok := client.(dynamic.Interface); !ok {
+					t.Errorf("got: %T, want: dynamic.Interface", client)
+				}
+			}
+		})
+	}
+
+}
+
+func TestAvailablePackageSummaryFromChart(t *testing.T) {
+	chartOK := &models.Chart{
+		Name:        "foo",
+		ID:          "foo",
+		Category:    "cat1",
+		Description: "best chart",
+		Icon:        "foo.bar/icon.svg",
+		Repo: &models.Repo{
+			Name:      "bar",
+			Namespace: "my-ns",
+		},
+		ChartVersions: []models.ChartVersion{
+			{Version: "1.0.0", AppVersion: "0.1.0"},
+			{Version: "1.0.0", AppVersion: "whatever"},
+			{Version: "whatever", AppVersion: "1.0.0"},
+		},
+	}
+
+	availablePackageSummaryOK := &corev1.AvailablePackageSummary{
+		DisplayName:      "foo",
+		LatestVersion:    "1.0.0",
+		IconUrl:          "foo.bar/icon.svg",
+		ShortDescription: "best chart",
+		AvailablePackageRef: &corev1.AvailablePackageReference{
+			Context:    &corev1.Context{Namespace: "my-ns"},
+			Identifier: "foo",
+		},
+	}
+
+	invalidChart := &models.Chart{Name: "foo"}
+
+	testCases := []struct {
+		name       string
+		in         *models.Chart
+		expected   *corev1.AvailablePackageSummary
+		statusCode codes.Code
+	}{
+		{
+			name:       "it returns AvailablePackageSummary if the chart is correct",
+			in:         chartOK,
+			expected:   availablePackageSummaryOK,
+			statusCode: codes.OK,
+		},
+		{
+			name:       "it returns internal error if empty chart",
+			in:         &models.Chart{},
+			statusCode: codes.Internal,
+		},
+		{
+			name:       "it returns internal error if chart is invalid",
+			in:         invalidChart,
+			statusCode: codes.Internal,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			availablePackageSummary, err := AvailablePackageSummaryFromChart(tc.in)
+
+			if got, want := status.Code(err), tc.statusCode; got != want {
+				t.Fatalf("got: %+v, want: %+v, err: %+v", got, want, err)
+			}
+
+			if tc.statusCode == codes.OK {
+				opt1 := cmpopts.IgnoreUnexported(corev1.AvailablePackageSummary{}, corev1.AvailablePackageReference{}, corev1.Context{})
+				if got, want := availablePackageSummary, tc.expected; !cmp.Equal(got, want, opt1) {
+					t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got, opt1))
+				}
+			}
+		})
+	}
+}
+
+func TestGetAvailablePackageSummaries(t *testing.T) {
+	getter := func(context.Context) (dynamic.Interface, error) {
```

Comment:
Not sure the k8s client is used currently, but it will be when we add a test ensuring that it 403s for namespaces to which the user doesn't have access, etc.